### PR TITLE
Misc additions from time aboard

### PR DESCRIPTION
**Pull request by hjabbot**
# PolarRoute Pull Request Template

Date: 2024-01-23
Version Number:  0.3.10

## Description of change
A collection of minor changes @gecoombs and I worked on while aboard the SDA.
- Moves waypoint from inaccessible cells to nearest accessible point
- Added ability to import/export in different file formats (GPX, ChartTracker-formatted-CSV's)
- Extract routes into individual GeoJSON's via CLI

## Fixes (various)

# Testing
To ensure that the functionality of the PolarRoute codebase remains consistent throughout the development cycle a testing strategy has been developed, which can be viewed in the document `test/testing_strategy.md`. 
This includes a collection of test files which should be run according to which part of the codebase has been altered in a pull request. Please consult the testing strategy to determine which tests need to be run. 

- [ ] My changes have not altered any of the files listed in the testing strategy

- [x] My changes result in all required regression tests passing without the need to update test files.  
  
> polar_route/cli.py
> polar_route/route_calc.py
> polar_route/route_planner.py
> polar_route/utils.py

[Passing Reg Tests](https://github.com/antarctica/PolarRoute/files/14025787/pr_pytest.txt)

- [ ] My changes require one or more test files to be updated for all regression tests to pass.   

# Checklist

- [x] My code follows [pep8](https://peps.python.org/pep-0008/) style guidelines.  
- [x] I have commented my code, particularly in hard-to-understand areas.  
- [x] I have updated the documentation of the codebase where required.  
- [x] My changes generate no new warnings.   
- [ ] My PR has been made to the `0.3.x` branch (**DO NOT SUBMIT A PR TO MAIN**)

### CODE REVIEW

File: polar_route/route_planner.py [hunk 1]
```diff
@@ -123,10 +122,62 @@ def _pandas_dataframe_str(input):
...
+    lat_max = mesh['config']['mesh_info']['region']['lat_max']
+    long_min = mesh['config']['mesh_info']['region']['long_min']
+    long_max = mesh['config']['mesh_info']['region']['long_max']
+    p1 = Point([long_min, lat_min])
+    p2 = Point([long_min, lat_max])
+    p3 = Point([long_max, lat_max])
+    p4 = Point([long_max, lat_min])
+    return Polygon([p1,p2,p3,p4])
+
+def _adjust_waypoints(point, cellboxes, max_distance=5):
```
Comment (SamuelHall700) [inline polar_route/route_planner.py:139]: Is there any reason this function is outside of the 'RoutePlanner' object?

**Review by SamuelHall700 — APPROVED** [review]
Looks good, addtional docstrings would be useful in a few places to help with the merge ahead of the route planning refactor for the 0.3.x release. Happy for this to be merged as-is though

**Comment by gecoombs** [comment]
It probably makes sense to add at least some documentation for these changes before merging. I'm happy to look at adding the new cli command to the docs.

**Review by gecoombs — CHANGES_REQUESTED** [review]
In order to facilitate the route planner refactor we agreed the flexible waypoints code should removed before merging.

**Review by gecoombs — APPROVED** [review]
Change of plan, we'll keep the flexible waypoint feature and resolve any conflicts with the refactor branch as part of that merge.

**Pull request merged**